kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Config.java

```diff
@@ -59,6 +60,8 @@
   public static final String KUBERNETES_ROLLING_TIMEOUT_SYSTEM_PROPERTY = "kubernetes.rolling.timeout";
   public static final String KUBERNETES_LOGGING_INTERVAL_SYSTEM_PROPERTY = "kubernetes.logging.interval";
 
+  // avoid inlining by calling new String("") in order to change value with reflection for test purposes
+  public static final String KUBERNETES_NAMESPACE_PATH = new String("/var/run/secrets/kubernetes.io/serviceaccount/namespace");
```

Comment:
Yeah there's no reason this couldn't be overridden, even if the default is probably going to be used 99.9% of the time.